File: 236/python/solutions.py
from typing import Optional

from lib.trees import BinaryTree
from nodes import BinaryTreeNode


class Solution1:
    def lowestCommonAncestor(self, root: BinaryTreeNode, p: BinaryTreeNode, q: BinaryTreeNode) -> \
            Optional[BinaryTreeNode]:
        return BinaryTree(root).lowestCommonAncestor(p, q)

File: 236/cpp/solutions.cpp
#include "lib/trees.h"
#include "nodes.h"

using lib::BinaryTree;
using dsa::nodes::BinaryTreeNode;

class Solution
{
public:
    BinaryTreeNode *lowestCommonAncestor(BinaryTreeNode *root, BinaryTreeNode *p, BinaryTreeNode *q)
    {
        BinaryTree *bt;
        bt = new BinaryTree(root);
        return bt->lowestCommonAncestor(p, q);
    }
};